server/models/service-instance.js

```diff
@@ -0,0 +1,70 @@
+var async = require('async');
+var debug = require('debug')('MeshModels.server.ServiceInstance');
+
+module.exports = function(ServiceInstance) {
+  ServiceInstance.beforeRemote(
+    'prototype.updateAttributes',
+    function(ctx, _, next) {
+      debug('updateAttributes with %j', ctx.args.data);
+      // When updating the instance via REST, only allow changes to cpus
+      ctx.args.data = {
+        cpus: ctx.args.data.cpus
+      };
+      next();
+    }
+  );
+
+  ServiceInstance.observe('before save', function beforeUpdate(ctx, next) {
```

Comment:
observe string "before save"... what's going on? does LB have two generations of ways of supplying hooks? can we use one way, or do we have to use the different ways?